2018-11-04  Ricky Ho  <classyho@lnxsrv10.seas.ucla.edu>

        * timespec.h (timespec_cmp): changed the implementation by 
	making comparisons between a.tv_sec and b.tv_sec to determine
	the return value, rather than returning the difference. 

--- lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ lib/timespec.h	2018-11-04 15:46:23.358661000 -0800
@@ -45,8 +45,12 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec < b.tv_sec)
+     return -1;
+  else if (a.tv_sec == b.tv_sec)
+     return 0; 
+  else
+     return 1;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
